scripts/plotting/global_mean_timeseries.py

```diff
@@ -0,0 +1,199 @@
+"""Use time series files to produce global mean time series plots for ADF web site.
+
+Includes a minimal Class for bringing CESM2 LENS data 
+from I. Simpson's directory (to be generalized).
+
+"""
+
+from pathlib import Path
+import warnings  # use to warn user about missing files.
+import xarray as xr
+import matplotlib.pyplot as plt
+import plotting_functions as pf
+
+
+def my_formatwarning(msg, *args, **kwargs):
+    """custom warning"""
+    # ignore everything except the message
+    return str(msg) + "\n"
+
+
+warnings.formatwarning = my_formatwarning
+
+
+def global_mean_timeseries(adfobj):
+    """
+    load time series file, calculate global mean, annual mean
+    for each case
+    Make a combined plot, save it, add it to website.
+    Include the CESM2 LENS result if it can be found.
+    """
+
+    plot_loc = get_plot_loc(adfobj)
+
+    plot_type = adfobj.read_config_var("diag_basic_info").get("plot_type", "png")
+
+    for field in adfobj.diag_var_list:
+        # reference time series (DataArray)
+        ref_ts_da = adfobj.data.load_reference_timeseries_da(field)
+        # check if this is a "2-d" varaible:
+        has_lat_ref, has_lev_ref = pf.zm_validate_dims(ref_ts_da)
```

Comment:
If the field doesn't exist (for some reason), the adfobj.data.load_reference_timeseries_da(field) will return a Nonetype, and 
  has_lat_ref, has_lev_ref = pf.zm_validate_dims(ref_ts_da)
fails, and ADF crashes. 

I added 
ref_ts_da = adfobj.data.load_reference_timeseries_da(field)
        if ref_ts_da is None:
            print(
                f"\t Variable named {field} provides Nonetype. Skipping this variable"
            )
            continue
        # check if this is a "2-d" varaible:
        has_lat_ref, has_lev_ref = pf.zm_validate_dims(ref_ts_da)
        
        which seems to work

Reply:
@adagj You're right, thanks for finding this!  I can confirm this fix works

Additionally, if the ADF is comparing against obs, it will also trigger the `Nonetype`.

@brianpm We can just add this check and not have the time series plot made if comparing against obs, or I have coded up a quick fix to allow the time series plots to be created in the model vs obs case, but will just plot the test time series. I'm happy to share if this desired.

Reply:
![Screenshot 2024-11-18 at 3 13 33 PM (3)](https://github.com/user-attachments/assets/f790a614-b953-4690-920b-bd07fbccc010)

Reply:
@justin-richling -- I think your solution to keep making the plot (just skip the reference) is the desired behavior. Can you incorporate your fix and merge it in?

Reply:
@brianpm Will do!